src/components/element-editor/prefill/prefill.tsx

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useState } from "react";
+import { MultiEntry } from "src/components/common";
+import { IPrefill, useBuilder } from "src/context-providers";
+import { PrefillChild } from "./prefill-child";
+import { useFormContext } from "react-hook-form";
+import { IBaseTextBasedFieldValues } from "src/schemas";
+
+export const Prefill = () => {
+    // =========================================================================
+    // CONST, STATES, REFS
+    // =========================================================================
+    const { focusedElement, updateFocusedElement } = useBuilder();
+    const element = focusedElement?.element;
+    const [childEntryValues, setChildEntryValues] = useState<IPrefill[]>([]);
+    const {
+        setValue,
+        formState: { isDirty },
+    } = useFormContext<IBaseTextBasedFieldValues>();
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+
+    const handleChildChange = (index: number, newValue: IPrefill) => {
+        setChildEntryValues((prevValues) => {
+            const updatedValues = [...prevValues];
+            updatedValues[index] = newValue;
+            setValue("prefill", updatedValues);
```

Comment:
You already have an effect function that updates already. There is no need to redo it here.

Reply:
if i remove line 28, the updating of the fields is a little slow and it does this 😅
![image](https://github.com/LifeSG/web-form-builder/assets/103546625/3561e58d-f1ff-4d5d-be5c-618f388c66bd)

Reply:
Hmm alright then. It feels a little weird to update the context in a state update. But i guess that is fine if it is to solve a specific issue.